_pytest/python.py

```diff
@@ -1132,6 +1132,10 @@ def __init__(self, name, parent, args=None, config=None,
         #: .. versionadded:: 3.0
         self.originalname = originalname
 
+        #: user properties is a list of tuples (name, value) that holds user
+        #: defined properties for this test.
+        self.user_properties = []
```

Comment:
we either need to add this in more places, or default to a empty list when trying to fetch it (i beleive its a good idea to move it to the item class)

Reply:
Thanks for the idea :)

I just added a commit with that change